pkg/sdk/poc/generator/resource_helper_methods.go

```diff
@@ -0,0 +1,122 @@
+package generator
+
+import (
+	"fmt"
+	"log"
+	"slices"
+)
+
+type objectIdentifier string
+
+const (
+	AccountObjectIdentifier             objectIdentifier = "AccountObjectIdentifier"
+	DatabaseObjectIdentifier            objectIdentifier = "DatabaseObjectIdentifier"
+	SchemaObjectIdentifier              objectIdentifier = "SchemaObjectIdentifier"
+	SchemaObjectIdentifierWithArguments objectIdentifier = "SchemaObjectIdentifierWithArguments"
+)
+
+func identifierStringToObjectIdentifier(s string) objectIdentifier {
+	switch s {
+	case "AccountObjectIdentifier":
+		return AccountObjectIdentifier
+	case "DatabaseObjectIdentifier":
+		return DatabaseObjectIdentifier
+	case "SchemaObjectIdentifier":
+		return SchemaObjectIdentifier
+	case "SchemaObjectIdentifierWithArguments":
+		return SchemaObjectIdentifierWithArguments
+	default:
+		return ""
+	}
+}
+
+type ResourceHelperMethodKind uint
+
+const (
+	ResourceIDHelperMethod ResourceHelperMethodKind = iota
+	ResourceObjectTypeHelperMethod
+)
+
+type ResourceHelperMethod struct {
+	Name        string
+	StructName  string
+	ReturnValue string
+	ReturnType  string
+}
+
+func newResourceHelperMethod(name, structName, returnValue string, returnType string) *ResourceHelperMethod {
+	return &ResourceHelperMethod{
+		Name:        name,
+		StructName:  structName,
+		ReturnValue: returnValue,
+		ReturnType:  returnType,
+	}
+}
+
+var requiredFieldsForIDMethodMapping map[objectIdentifier][]string = map[objectIdentifier][]string{
+	AccountObjectIdentifier:  {"Name"},
+	DatabaseObjectIdentifier: {"Name", "DatabaseName"},
+	SchemaObjectIdentifier:   {"Name", "DatabaseName", "SchemaName"},
+}
+
+func hasRequiredFieldsForIDMethod(structName string, helperStructs []*Field, requiredFields ...string) bool {
```

Comment:
Nit: we should receive `requiredFields ...objectIdentifier` and convert to `string` inside.